modules/AlphaWalletCore/AlphaWalletCore/URL+Extensions.swift

```diff
@@ -16,11 +16,15 @@ extension URL {
             }()
             return URL(string: urlString)
         } else {
+            guard let components = URLComponents(url: self, resolvingAgainstBaseURL: false) else { return nil }
+            if components.host == "ipfs.infura.io" {
+                return URL(string: "https://alphawallet.infura-ipfs.io" + components.path)
```

Comment:
I wonder if it is better just do a search and replace for "https://ipfs.infura.io/" once the host check passes. It breaks *if* the URL also includes that string outside the host (which seems unlikely or impossible?), but otherwise it works with URLS that have query strings. Maybe with anchors too? Anchors are probably unlikely, but query strings seems possible.

eg. the "?x=a" in "https://ipfs.infura.io/ipfs/abc?x=a"

Unless there's a better way than `components.path`?

Reply:
updated pr replacing host and scheme, so path and params will be the same as in source